File: webapp/src/app/equipments/model/equipment.model.ts
import {MabecCode} from "./mabec-code.model";
import {ProtectionType} from "./protection-type.model";

export class EquipmentModel {
  constructor(public id: number,
              public name: string,
              public code: string,
              public mabecCode: MabecCode,
              public protectionType: ProtectionType,
              public size: string,
              public sex: string,
              public startDate: Date,
              public endDate: Date,
              public expirationDate:Date,
              public isChecked: boolean = false
  ) {}
}

File: src/main/java/ro/esolutions/eipl/models/EquipmentModel.java
package ro.esolutions.eipl.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import ro.esolutions.eipl.types.MabecCode;
import ro.esolutions.eipl.types.ProtectionType;

import javax.validation.constraints.NotNull;
import java.time.LocalDate;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class EquipmentModel {
    private Long id;
    @NotNull
    private String name;
    @NotNull
    private String code;

    @NotNull
    private MabecCode mabecCode;

    private ProtectionType protectionType;
    @NotNull
    private String size;
    @NotNull
    private String sex;

    private LocalDate expirationDate;
}